xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/EditAction.java

```diff
@@ -106,7 +106,7 @@ protected XWikiDocument prepareEditedDocument(XWikiContext context) throws XWiki
 
         // The default values from the template can be overwritten by additional request parameters.
         updateDocumentTitleAndContentFromRequest(editedDocument, context);
-        editedDocument.readObjectsFromForm(editForm, context);
+        editedDocument.readAddedUpdatedAndRemovedObjectsFromForm(editForm, context);
```

Comment:
Not sure to get it: thought the bug was in PreviewAction?

Reply:
`PreviewAction` inherits from `EditAction` and uses this `prepareEditedDocument` method to get the edited document. The same bug also affects the edit action when you, e.g., return from preview to edit.